integrations/extensions/docs/elasticsearch-install-and-setup/text_embedding_deploy_and_use.md

```diff
@@ -0,0 +1,189 @@
+        ...
+```
+
+## Load sample data
+Refer to the [Load data into Elasticsearch](./ICD_Elasticsearch_install_and_setup.md#load-data-into-elasticsearch) section in the Elasticsearch setup guide to upload a sample data to Elasticsearch using Kabana.
+
+## Add your embedding model to an inference ingest pipeline
+Create an ingest pipeline called `text-embeddings` using the command below:
+```bash
+curl -X PUT "${ES_URL}/_ingest/pipeline/text-embeddings" \
+  -u "${ES_USER}:${ES_PASSWORD}" --cacert "${ES_CACERT}"\
+  -H 'Content-Type: application/json' -d '{
+  "description": "Text embedding pipeline",
+  "processors": [
+    {
+      "inference": {
+        "model_id": "intfloat__multilingual-e5-small",
+        "target_field": "text_embedding",
+        "field_map": {
+          "text": "text_field"
+        }
+      }
+    }
+  ],
+  "on_failure": [
+    {
+      "set": {
+        "description": "Index document to '\''failed-<index>'\''",
+        "field": "_index",
+        "value": "failed-{{{_index}}}"
+      }
+    },
+    {
+      "set": {
+        "description": "Set error message",
+        "field": "ingest.failure",
+        "value": "{{_ingest.on_failure_message}}"
+      }
+    }
+  ]
+}'
+```
+
+You can verify that the `text-embeddings` ingest pipeline was created by locating it in the list of your ingest pipelines on Kibana http://localhost:5601/app/management/ingest/ingest_pipelines
+
+## Create a mapping for the destination index containing the embeddings
+Then run the command below to create the mappings of the destination index called `wa-docs-with-embeddings`:
+```bash
+curl -X PUT "${ES_URL}/wa-docs-with-embeddings" \
+  -u "${ES_USER}:${ES_PASSWORD}" --cacert $ES_CACERT \
+  -H 'Content-Type: application/json' -d '{
+  "mappings": {
+    "properties": {
+      "text_embedding.predicted_value": {
+        "type": "dense_vector",
+        "dims": 384,
+        "index": true,
+        "similarity": "cosine"
+      },
+      "text": {
+        "type": "text"
+      }
+    }
+  }
+}'
+```
+
+* `text_embedding.predicted_value` is the field where the ingest processor stores the embeddings
+* `dims` is the embedding size of the deployed model which is 384 for the `intfloat/multilingual-e5-small` model we are using here
+
+## Create the text embeddings
+Run the `text-embeddings` ingest pipeline to reindex the data to the `wa-docs-with-embeddings` index
+```bash
+curl -X POST "${ES_URL}/_reindex?wait_for_completion=false" \
+  -u "${ES_USER}:${ES_PASSWORD}" --cacert $ES_CACERT \
+  -H 'Content-Type: application/json' -d '{
+  "source": {
+    "index": "wa-docs",
+    "size": 50
+  },
+  "dest": {
+    "index": "wa-docs-with-embeddings",
+    "pipeline": "text-embeddings"
+  }
+}'
+```
+
+This command will return a task id that looks like this:
+```json
+{"task":<task-id>}
+```
+
+The reindexing process can take around 10 minutes. You can use the task id that is returned above to check the status of the process.
```

Comment:
Is there a specific part of the response of the `_tasks` call that the reader should look for?  I guess it's just `"completed":true` ?

Reply:
Yes, that's correct. I'll add a note pointing that out.